src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -944,6 +946,10 @@ subroutine shelf_calc_flux(sfc_state_in, fluxes_in, Time, time_step_in, CS)
   if (CS%id_shelf_mass > 0) call post_data(CS%id_shelf_mass, ISS%mass_shelf, CS%diag)
   if (CS%id_area_shelf_h > 0) call post_data(CS%id_area_shelf_h, ISS%area_shelf_h, CS%diag)
   if (CS%id_ustar_shelf > 0) call post_data(CS%id_ustar_shelf, fluxes%ustar_shelf, CS%diag)
+  if (allocated(sfc_state%taux_shelf) .and. allocated(sfc_state%tauy_shelf)) then
+    if (CS%id_taux > 0) call post_data(CS%id_taux, sqrt(taux2), CS%diag)
```

Comment:
We usually discourage the use of array-syntax math in arguments, as would be the case here with `sqrt(taux2)`, as described at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#array-syntax, because it can have unintended results when working in halos that may not have been properly initialized.  In this case, the halos have all been set to 0, but once we allow this construct it can have dangerous consequences in other cases that are not coded so carefully.